internal/scorecard/alpha/formatting.go

```diff
@@ -42,17 +42,17 @@ func (r PodTestRunner) getTestStatus(ctx context.Context, p *v1.Pod) (output *v1
 // run based on user selection
 func (o Scorecard) List() v1alpha3.TestList {
 	output := v1alpha3.NewTestList()
-	tests := o.selectTests()
-
-	for _, test := range tests {
-		item := v1alpha3.NewTest()
-		item.Spec = v1alpha3.TestSpec{
-			Image:      test.Image,
-			Entrypoint: test.Entrypoint,
-			Labels:     test.Labels,
+	for _, stage := range o.Config.Stages {
+		tests := o.selectTests(stage)
+		for _, test := range tests {
+			item := v1alpha3.NewTest()
+			item.Spec = v1alpha3.TestSpec{
+				Image:      test.Image,
+				Entrypoint: test.Entrypoint,
+				Labels:     test.Labels,
+			}
+			output.Items = append(output.Items, item)
```

Comment:
usually no happy with for inside to another for, however, I think it will not have too many items so 👍

Reply:
That's the only way this works :smile: 

This isn't an O(n^2) problem like it seems like you're referring to. Stages are just groups of tests. This function is (generally) O(_number of tests_) regardless of whether stages are introduced or not.